shm/Ship.cpp

```diff
@@ -1,35 +1,31 @@
+#include "Cargo.hpp"
 #include "Ship.hpp"
-    
-Ship::Ship() 
-    : id_(-1)
-{}
 
-Ship::Ship(int capacity, int maxCrew, int speed, const std::string& name, size_t id, std::vector<std::unique_ptr<Cargo>> cargoVec)
-    : capacity_(capacity)
-    , maxCrew_(maxCrew)
-    , crew_(0)
-    , speed_(speed)
-    , name_(name)
-    , id_(id)
-    , cargoVec_(std::move(cargoVec))
+Ship::Ship(size_t capacity, size_t maxCrew, size_t speed, const std::string& name, size_t id, CargoVec cargoVec)
+    : capacity_(capacity), maxCrew_(maxCrew), crew_(0), speed_(speed), name_(name), id_(id), cargoVec_(std::move(cargoVec))
 {}
 
-Ship::Ship(int maxCrew, int speed, size_t id)
+Ship::Ship(size_t maxCrew, size_t speed,const size_t id)
```

Comment:
const with size_t is not neccessary. Because this is a copy of the original value.